crates/core/src/protobuf.rs

```diff
@@ -34,9 +38,133 @@ pub mod node_ctl_svc {
 }
 
 pub mod metadata_proxy_svc {
+    use std::sync::Arc;
+
+    use bytestring::ByteString;
+    use tokio::sync::Mutex;
+    use tonic::{codec::CompressionEncoding, transport::Channel};
+
+    use metadata_proxy_svc_client::MetadataProxySvcClient;
+    use restate_types::{
+        Version,
+        metadata::{Precondition, VersionedValue},
+        nodes_config::NodesConfiguration,
+    };
+
+    use crate::metadata_store::{MetadataStore, ProvisionError, ReadError, WriteError};
+
+    use super::{to_read_err, to_write_err};
 
     tonic::include_proto!("restate.metadata_proxy_svc");
 
     pub const FILE_DESCRIPTOR_SET: &[u8] =
         tonic::include_file_descriptor_set!("metadata_proxy_svc_descriptor");
+
+    pub struct MetadataStoreProxy {
+        client: Arc<Mutex<MetadataProxySvcClient<Channel>>>,
```

Comment:
Instead of `Arc<Mutex<>>` could we use `MetadataProxySvcClient<Channel>` directly? Cloning this struct should be cheap according to the docs of `Channel`.

Reply:
The problem is MetadataProxySvcClient calls all require exclusive reference, but the MetadataStore only provides a shared reference. Otherwise, i would have implemented MetadataStore directly on the `MetdataProxySvcClient`

Reply:
Would it be possible to clone `MetadataProxySvcClient` for the calls that require an exclusive reference?

Reply:
Yes 🤦🏼‍♂️